packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasSummaryStore.ts

```diff
@@ -0,0 +1,234 @@
+            })
+
+            setResult(estimateGasResult)
+          }
+        } else {
+          if (token) {
+            let estimateGasResult: {
+              estimatedL1Gas: BigNumber
+              estimatedL2Gas: BigNumber
+            }
+
+            if (
+              isTokenArbitrumOneNativeUSDC(token.address) ||
+              isTokenArbitrumGoerliNativeUSDC(token.address)
+            ) {
+              estimateGasResult = {
+                estimatedL1Gas: constants.Zero,
+                estimatedL2Gas: constants.Zero
+              }
+              setStatus('unavailable')
+              return
+            } else {
+              estimateGasResult = await withdrawTokenEstimateGas({
+                amount: amountDebounced,
+                erc20L1Address: token.address,
+                address: walletAddress,
+                l2Provider: l2.provider
+              })
+            }
+
+            setResult({
+              ...estimateGasResult,
+              estimatedL2SubmissionCost: constants.Zero
+            })
+          } else {
+            const estimateGasResult = await withdrawEthEstimateGas({
+              amount: amountDebounced,
+              address: walletAddress,
+              l2Provider: l2.provider
+            })
+
+            setResult({
+              ...estimateGasResult,
+              estimatedL2SubmissionCost: constants.Zero
+            })
+          }
+        }
+
+        setStatus('success')
+      } catch (error) {
+        console.error(error)
+        setStatus('error')
+      }
+    }
+
+    if (arbTokenBridge && arbTokenBridge.eth && arbTokenBridge.token) {
+      estimateGas()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    // Re-run gas estimation when:
+    isDepositMode, // when user switches deposit/withdraw mode
+    amount, // when user changes the amount (check against the debounced value)
+    amountDebounced,
+    token, // when the token changes
+    shouldRunGasEstimation, // passed externally - estimate gas only if user balance crosses a threshold
+    l1.network.id, // when L1 and L2 network id changes
+    l2.network.id,
+    walletAddress // when user switches account or if user is not connected
+  ])
+
+  const gasSummaryResult = {
+    status,
+    estimatedL1GasFees,
+    estimatedL2GasFees
+  }
+
+  // rather than a deep equal, we are checking all the properties to see if they are sync
+  if (
+    gasSummary.status !== status ||
+    (gasSummary.estimatedL1GasFees !== estimatedL1GasFees &&
+      gasSummary.estimatedL2GasFees !== estimatedL2GasFees)
+  ) {
+    setGasSummary(gasSummaryResult)
+  }
+}
+
+type GasSummaryStore = {
+  gasSummary: UseGasSummaryResult
+  setGasSummary: (gasSummary: UseGasSummaryResult) => void
+}
+
+export const useGasSummaryStore = create<GasSummaryStore>()(set => ({
+  gasSummary: INITIAL_GAS_SUMMARY_RESULT,
+  setGasSummary: gasSummary => set(() => ({ gasSummary }))
+}))
```

Comment:
I'm not sure I understand the value of the store here? If it's to share the state among multiple hook calls, then can we use SWR? But maybe it would be too big of a change for this PR.